src/main/scala/Divider.scala

```diff
@@ -76,26 +77,11 @@ object Divider:
 
       case _ => 0
 
-  private val mixednessRegions: List[List[Square]] = {
+  private def mixedness(board: Board): Int = {
+    val smallSquare = 0x0303L.bb
     for
```

Comment:
I just wondering if we can gain any performance by move this calculation out from `mixedness` would help?:

```scala
val regions = 
  val val smallSquare = 0x0303L.bb
  for
      y <- 0 to 6
      x <- 0 to 6
  yield smallSquare << (x + 8 * y)
```
and then
```scala
private def mixedness(board: Board): Int =
  regions.fold(0): (acc, region) =>
    acc + board.byColor.map(c => (c & region).count).reduce(score(y + 1))
```

Reply:
Unfortunately `y` comes from up there. `region.first.get.rank.index + 1` would work but is a bit ugly.

Reply:
Looks like your intuition is right, though, and it's a quite significant optimization:

```
[info] PlayBench.divider  thrpt   45  51.566 ± 4.220  ops/s
```

Reply:
ah, I missed that `y`, so then I have another idea :D,
we can returns the y in the list of regions:
```scala
val regions = 
  val smallSquare = 0x0303L // also I guess we don't need bitboard here.
  for
      y <- 0 to 6
      x <- 0 to 6
  yield (smallSquare << (x + 8 * y), y)
  
private def mixedness(board: Board): Int =
  regions.fold(0): (acc, (region, y)) =>
    acc + board.byColor.map(c => (c & region).count).reduce(score(y + 1))
```